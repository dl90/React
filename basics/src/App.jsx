import React from 'react'
import logo from './logo.svg'
import './App.css'

function App (props) {
  const { name, email, ...others } = props.data
  props.children(others)

  let message
  (!name || !email)
    ? message = 'Ran into an error'
    : message = `Hello ${name}`

  return (
    <div className='App'>
      <header className='App-header'>
        <img src={logo} className='App-logo' alt='logo' />
        <h1>
          {message}
        </h1>
        {email ?? ''}
      </header>

      <ListOfTenThings />
    </div>
  )
}

// Calls the children callback numTimes to produce a repeated component
function Repeat (props) {
  const items = []
  for (let i = 0; i < props.numTimes; i++) {
    items.push(props.children(i))
  }
  return <div>{items}</div>
}

function ListOfTenThings () {
  return (
    <Repeat numTimes={10}>
      {(index) => <div key={index + 1}>This is item {index + 1} in the list</div>}
    </Repeat>
  )
}

/* component class declaration */
// class App extends React.Component {
//   render () {
//     const { name, email } = this.props.data

//     return (
//       <div className='App'>
//         <header className='App-header'>
//           <img src={logo} className='App-logo' alt='logo' />
//           <h1>
//             Hello {name}
//           </h1>
//           {email}
//         </header>
//       </div>
//     )
//   }
// }

export { App }
