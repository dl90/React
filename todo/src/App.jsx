import React from 'react'
import { BrowserRouter, Route, Switch } from 'react-router-dom'

import Navbar from './layout/Navbar'
import Home from './pages/Home'
import About from './pages/About'
import Error from './pages/Error'

export default function App () {
  const LINKS = {
    home: '/',
    about: '/about'
  }
  return (
    <main className='todoapp stack-large'>
      <BrowserRouter>
        <Navbar {...LINKS} />
        <Switch>
          <Route path={LINKS.home} component={Home} exact />
          <Route path={LINKS.about} component={About} />
          <Route component={Error} />
        </Switch>
      </BrowserRouter>
    </main>
  )
}
