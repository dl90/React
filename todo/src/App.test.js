/* global describe test expect */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import App from './App'

describe('main', () => {
  test('renders', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
  })

  test('snapShot', () => {
    const component = renderer.create(<App />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  })
})
