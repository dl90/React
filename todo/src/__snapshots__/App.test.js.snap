// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main snapShot 1`] = `
<div
  className="todoapp stack-large"
>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </div>
  <div>
    <h1>
      <span
        aria-label="tic-toc"
        role="img"
      >
        üí£üí•üèÅ
      </span>
    </h1>
    <form
      data-testid="form"
      onSubmit={[Function]}
    >
      <h2
        className="label-wrapper"
      >
        <label
          className="label__lg"
          htmlFor="new-todo-input"
        >
          What needs to be done?
        </label>
      </h2>
      <input
        autoComplete="off"
        className="input input__lg"
        id="new-todo-input"
        name="text"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="btn btn__primary btn__lg"
        type="submit"
      >
        Add
      </button>
    </form>
    <div
      className="btn-group"
    >
      <button
        className="btn btn__danger"
        onClick={[Function]}
        type="button"
      >
        <span>
          Clear All
        </span>
      </button>
    </div>
    <div
      className="filters btn-group stack-exception"
    >
      <button
        aria-pressed={true}
        className="btn toggle-btn"
        onClick={[Function]}
        type="button"
      >
        <span
          className="visually-hidden"
        >
          Show 
        </span>
        <span
          data-testid="filterButtonName"
        >
          All
        </span>
        <span
          className="visually-hidden"
        >
           tasks
        </span>
      </button>
      <button
        aria-pressed={false}
        className="btn toggle-btn"
        onClick={[Function]}
        type="button"
      >
        <span
          className="visually-hidden"
        >
          Show 
        </span>
        <span
          data-testid="filterButtonName"
        >
          Active
        </span>
        <span
          className="visually-hidden"
        >
           tasks
        </span>
      </button>
      <button
        aria-pressed={false}
        className="btn toggle-btn"
        onClick={[Function]}
        type="button"
      >
        <span
          className="visually-hidden"
        >
          Show 
        </span>
        <span
          data-testid="filterButtonName"
        >
          Completed
        </span>
        <span
          className="visually-hidden"
        >
           tasks
        </span>
      </button>
    </div>
    <h2
      id="list-heading"
    >
      0 tasks remaining
    </h2>
    <ul
      aria-labelledby="list-heading"
      className="todo-list stack-large stack-exception"
    />
  </div>
</div>
`;
